third_party/move/testing-infra/transactional-test-runner/src/vm_test_harness.rs

```diff
@@ -150,31 +167,81 @@
             default_syntax,
             comparison_mode,
             run_config,
-            storage: InMemoryStorage::new(),
+            vm,
+            resource_storage: InMemoryStorage::new(),
+            module_bytes_storage: LocalModuleBytesStorage::empty(),
         };
 
-        adapter
-            .perform_session_action(None, |session, gas_status| {
-                for module in either_or_no_modules(pre_compiled_deps_v1, pre_compiled_deps_v2)
-                    .into_iter()
-                    .map(|tmod| &tmod.named_module.module)
-                {
+        if vm_config.use_loader_v2 {
+            let addresses = either_or_no_modules(pre_compiled_deps_v1, pre_compiled_deps_v2)
```

Comment:
Was wondering why we need this logic here? IIUC we try to put the modules into the local store but was wondering can we just leave them as unresolved form and load them as we go?